fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryLockFactory.java

```diff
@@ -195,20 +200,21 @@ protected void fileLockClearIfLocked(boolean isRecovery) {
                                     UUID existingUuid = fileLockValueToUuid(val);
                                     if (existingUuid != null && existingUuid.compareTo(selfStampUuid) == 0) {
                                         aContext.ensureActive().clear(fileLockKey);
-                                        logSelf("Cleared Lucene File Lock in Recovery path");
+                                        logSelf("FileLock: Cleared in Recovery path");
                                     }
                                 }
                             });
 
-            if (isRecovery) {
-                // Here: this is called in the recovery path
+            if (agilityContext.isClosed()) {
+                // Here: this is considered to be a recovery path, may bypass closed context.
                 agilityContext.asyncToSync(LuceneEvents.Waits.WAIT_LUCENE_FILE_LOCK_CLEAR,
                         agilityContext.applyInRecoveryPath(fileLockFunc));
             } else {
                 // Here: this called during directory close to ensure cleared lock -
                 agilityContext.asyncToSync(LuceneEvents.Waits.WAIT_LUCENE_FILE_LOCK_CLEAR,
```

Comment:
I think if the AgilityContext fails to commit after this, it won't ever do the clear, which seems like a bug. Perhaps worth decoupling the logging, and `rebalance` -> `repartition` changes from this, and bringing this in with additional testing.

Reply:
If needed, `FDBDirectory.close()`  will retry to clear the lock.

Reply:
Fixed: agilityContext must succeed flush before marking the lock as closed. 
✅